acceptance/suites/tests/020-external-ca/010_setup_remote_ca.rb

```diff
@@ -0,0 +1,136 @@
+confine(:except, :platform => ['windows','aix','solaris','darwin'])
+
+# Pre-setup: Installing necessary packages
+step 'Install facter and OpenSSL on External CA' do
+  on(ca, "rpm -Uvh http://dl.fedoraproject.org/pub/epel/6/x86_64/epel-release-6-8.noarch.rpm", :acceptable_exit_codes => [0,1])
+  install_package ca, 'facter'
+  install_package ca, 'openssl'
+end
+
+#Pre-setup: Generate SSH Keys on Root CA
+if on(ca, "test ! -f /root/.ssh/id_rsa", :accept_all_exit_codes => true).exit_code == 0
+  on(ca, "ssh-keygen -f /root/.ssh/id_rsa -N \"\"")
+end
+ca_key = on(ca, "cat /root/.ssh/id_rsa.pub").stdout.chomp
+
+#Pre-setup: Get Hostnames and FQDNs of Puppet Master & Root CA
+pm_hostname = fact_on(master, "hostname").chomp
+pm_fqdn = fact_on(master, "fqdn").chomp
+pm_domain = fact_on(master, "domain").chomp
+ca_fqdn = fact_on(ca, "fqdn").chomp
+ca_hostname = fact_on(ca, "hostname").chomp
+
+#Pre-Setup: Define Directories used on Puppet Master and Root CA
+ca_dir = "/root/rakeca"
+int_ca_dir = "/root/rakeca/intermediate"
+p_ssl = "/etc/puppetlabs/puppet/ssl"
+
+step 'Push External CA public key onto all hosts' do
```

Comment:
Might be good to disambiguate this as an SSH user account public key rather than an SSL based public key.